packages/cli/generators/import-lb3-models/migrate-model.js

```diff
@@ -43,12 +44,29 @@ function importLb3ModelDefinition(modelCtor, log) {
 
   logNamingIssues(modelName, log);
 
+  const baseDefinition = modelCtor.base.definition;
+  const baseProps = {...baseDefinition.properties};
+
+  // Core LB3 models like PersistedModel come with an id property that's
+  // injected by juggler. We don't want to inherit that property, because
+  // in LB4, we want models to define the id property explicitly.
+  if (isCoreModel(modelCtor.base)) {
```

Comment:
Sorry I am a bit confused. Does this mean if a custom model inherits a base model, it won't have the id property after importing to LB4 app?

Reply:
When scaffolding code for an imported model, we want to skip all properties inherited from the base model. For example, if `User` model has `email` property and `Customer` model inherits from `User`, then the code for `Customer` SHOULD NOT re-define `email` property again. The property should be inherited instead, both at TypeScript and juggler-metadata level.

Now juggler's `PersistedModel` always comes with `id` property (in fact, the property is added to every model created by juggler, including the base `Model` class), see [ModelBuilder.prototype.define()](https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/model-builder.js#L311-L339).

As a result, when deciding which properties belong to the model being imported (e.g. `Product`) and which are inherited (e.g. from `PersistedModel`), the primary key property `id` was always skipped before I introduced this check, simply because it was considered as coming from the base model (`PersistedModel`).